uefi/flashregionsection.go

```diff
@@ -12,14 +13,26 @@ const FlashRegionSectionSize = 36
 
 // FlashRegionSection holds the metadata of all the different flash regions like PDR, Gbe and the Bios region.
 type FlashRegionSection struct {
-	Reserved            uint16
+	Reserved            uint16 `json:"-"`
 	FlashBlockEraseSize uint16
 	BiosBase, BiosLimit uint16
 	MeBase, MeLimit     uint16
 	GbeBase, GbeLimit   uint16
 	PdrBase, PdrLimit   uint16
 }
 
+type jsonFlashRegionSection FlashRegionSection
+// MarshalJSON marshals the FlashRegionSection into JSON
+func (f FlashRegionSection) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		AvailableRegions string `json:"Regions"`
+		jsonFlashRegionSection
+	}{
+		AvailableRegions : strings.Join(f.AvailableRegions(), ","),
```

Comment:
What's the point? You're just encoding the data twice in the JSON through both the Regions field and BiosLimit.

Reply:
I think if you're doing this for readability, all these things should just have a fmt.Stringer for the human-readable stuff (e.g. fmt.Sprintf("FlashRegionSecton{%v, AvailableRegions=%s}", f, f.AvailableRegions())). Let's not do that in JSON.

Reply:
we should separate human-readable strings from machine encoding.

Reply:
That's a fair point but I didn't want to have separate parsing rules for both of them. I was hoping we could just have the same json for both humans and machine encoding.

Should I then keep the original summary stuff?

Reply:
Maybe keep some of them (those where you're unhappy with %+v), and make them fmt.Stringer implementations instead of our own name.

Reply:
I guess the point I want to make is this: I'd like to see tests for the machine encoding with marshaling and unmarshaling. If the marshaling and unmarshaling is just json.Marshal/json.Unmarshal that pretty well covers it, because those are well-tested. But if we're to implement our own MarshalJSON and UnmarshalJSON methods, we have to test them. 

We don't have to really test the human-readable stuff. So if you separate it such that JSON stuff is *just* json.Marshal and json.Unmarshal, and human-readable stuff is whatever, no tests necessary.